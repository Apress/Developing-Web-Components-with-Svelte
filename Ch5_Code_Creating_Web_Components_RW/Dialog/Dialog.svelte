<svelte:options tag="cobalt-dialog" />

<script>
  import CloseIcon from "./Close.svelte";
  export let show = "true"; // THis is the one to change!
  let showDialog = show == "true";
</script>

<button on:click={() => (showDialog = !showDialog)}>Show dialog</button>

{#if showDialog}
  <div class="cobalt">
    <div id="background" />
    <div id="modal">
      <div class="header">
        <h3>Modal title</h3>
        <button
          type="button"
          class="close"
          on:click={() => (showDialog = false)}
        >
          <CloseIcon />
        </button>
      </div>
      <p>Click on the X to close me</p>
    </div>
  </div>
{/if}

<style>
  .cobalt {
    font-family: Arial, Helvetica, sans-serif;
  }

  @keyframes fadein {
    from {
      opacity: 0;
    }
    to {
      opacity: 1;
    }
  }

  #background {
    position: fixed;
    z-index: 1;
    top: 0;
    left: 0;
    width: 100vw;
    height: 100vh;
    background-color: rgba(0, 0, 0, 0.7);
    animation: fadein 0.5s;
  }

  #modal {
    position: fixed;
    z-index: 2;
    top: 50%;
    left: 50%;
    transform: translate(-50%, -50%);
    background: #fff;
    padding: 10px;
    width: 400px;
    height: 250px;
  }

  .header {
    display: flex;
    justify-content: space-between;
    border-bottom: 1px solid #c4c4c4;
  }

  #modal div button {
    display: contents;
    color: #19247c;
  }

  #modal div button:hover {
    color: #9393ff;
  }
</style>
