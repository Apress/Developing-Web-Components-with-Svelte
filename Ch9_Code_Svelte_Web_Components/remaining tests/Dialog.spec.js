import { render, screen } from '@testing-library/svelte';
import { fireEvent } from '@testing-library/dom';
import Dialog from "../src/lib/Dialog/Dialog.svelte";

describe("Tests for Dialog", () => {
  const props = { };

  it("should render properly", async() => {
    const result = render(Dialog, { props });
    expect(() => result).not.toThrow();
  });

  it("should show the dialog when clicking the button", async() => {
    const result = render(Dialog, { props });
    fireEvent.click(screen.getByTestId("buttonDialog"));
    expect(screen.getByRole("dialog")).tobeVisible;
  })

  it("should disappear when close button clicked", async() => {
    const result = render(Dialog, { props });
    fireEvent.click(screen.getByTestId("buttonDialog"));
    expect(screen.getByRole("dialog")).tobeVisible;
    fireEvent.click(screen.getByTestId("buttonDialog"));
    expect(screen.getByRole("dialog")).not.tobeVisible;
  })

  it("get a snapshot of component", () => {
    const tree = render(Dialog, props);
    expect(tree).toMatchSnapshot();
  })
});