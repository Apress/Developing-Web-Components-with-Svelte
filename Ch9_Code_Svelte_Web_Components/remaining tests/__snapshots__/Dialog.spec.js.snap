// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for Dialog get a snapshot of component 1`] = `
{
  "component": Dialog {
    "$$": {
      "after_update": [],
      "before_update": [],
      "bound": {},
      "callbacks": {},
      "context": Map {},
      "ctx": [
        true,
        "true",
        [Function],
        [Function],
      ],
      "dirty": [
        -1,
      ],
      "fragment": {
        "c": [Function],
        "d": [Function],
        "i": [Function],
        "l": [Function],
        "m": [Function],
        "o": [Function],
        "p": [Function],
      },
      "not_equal": [Function],
      "on_destroy": [
        [Function],
      ],
      "on_disconnect": [],
      "on_mount": [],
      "props": {
        "show": 1,
      },
      "root": <div>
        <button
          data-testid="buttonDialog"
        >
          Show dialog
        </button>
         
        <div
          class="cobalt svelte-18e6gsj"
        >
          <div
            class="svelte-18e6gsj"
            id="background"
          />
           
          <div
            class="svelte-18e6gsj"
            id="modal"
            role="dialog"
          >
            <div
              class="header svelte-18e6gsj"
            >
              <h3>
                Modal title
              </h3>
               
              <button
                class="close svelte-18e6gsj"
                type="button"
              >
                <button
                  aria-label="Close"
                  class="close-button svelte-myhoof"
                >
                  <svg
                    aria-hidden="true"
                    class="close"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M.439 21.44a1.5 1.5 0 0 0 2.122 2.121l9.262-9.261a.25.25 0 0 1 .354 0l9.262 9.263a1.5 1.5 0 1 0 2.122-2.121L14.3 12.177a.25.25 0 0 1 0-.354l9.263-9.262A1.5 1.5 0 0 0 21.439.44L12.177 9.7a.25.25 0 0 1-.354 0L2.561.44A1.5 1.5 0 0 0 .439 2.561L9.7 11.823a.25.25 0 0 1 0 .354Z"
                      fill="currentColor"
                    />
                  </svg>
                </button>
                
              </button>
            </div>
             
            <p>
              Click on the X to close me
            </p>
          </div>
        </div>
        
      </div>,
      "skip_bound": false,
      "update": [Function],
    },
    "$$set": [Function],
    "$capture_state": [Function],
    "$inject_state": [Function],
  },
  "container": <body>
    <div>
      <button
        data-testid="buttonDialog"
      >
        Show dialog
      </button>
       
      <div
        class="cobalt svelte-18e6gsj"
      >
        <div
          class="svelte-18e6gsj"
          id="background"
        />
         
        <div
          class="svelte-18e6gsj"
          id="modal"
          role="dialog"
        >
          <div
            class="header svelte-18e6gsj"
          >
            <h3>
              Modal title
            </h3>
             
            <button
              class="close svelte-18e6gsj"
              type="button"
            >
              <button
                aria-label="Close"
                class="close-button svelte-myhoof"
              >
                <svg
                  aria-hidden="true"
                  class="close"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M.439 21.44a1.5 1.5 0 0 0 2.122 2.121l9.262-9.261a.25.25 0 0 1 .354 0l9.262 9.263a1.5 1.5 0 1 0 2.122-2.121L14.3 12.177a.25.25 0 0 1 0-.354l9.263-9.262A1.5 1.5 0 0 0 21.439.44L12.177 9.7a.25.25 0 0 1-.354 0L2.561.44A1.5 1.5 0 0 0 .439 2.561L9.7 11.823a.25.25 0 0 1 0 .354Z"
                    fill="currentColor"
                  />
                </svg>
              </button>
              
            </button>
          </div>
           
          <p>
            Click on the X to close me
          </p>
        </div>
      </div>
      
    </div>
  </body>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
